hibernate-rx-core/src/main/java/org/hibernate/rx/service/RxConnectionPoolProviderImpl.java

```diff
@@ -59,10 +62,21 @@ private Pool configurePool(Map configurationValues) {
 
 		final String url = ConfigurationHelper.getString(AvailableSettings.URL, configurationValues);
 		final URI uri = JdbcUrlParser.parse( url );
-		final String database = uri.getPath().substring( 1 );
+		String database = uri.getPath().substring( 1 );
+		if (uri.getScheme().equals("db2") && database.indexOf( ':' ) > 0) {
+			database = database.substring( 0, database.indexOf( ':' ) );
+		}
 
 		if (username==null || password==null) {
-			String[] params = uri.getQuery().split("&");
+			String[] params = {};
+			if (uri.getScheme().equals("db2")) {
+			    int queryIndex = uri.getPath().indexOf(':') + 1;
+			    if (queryIndex > 0) {
+			      params = uri.getPath().substring(queryIndex).split(";");
+			    }
```

Comment:
An example of the db2 uri here would help to understand why you need the extra logic